packages/react-components/react-dialog/src/contexts/dialogContext.ts

```diff
@@ -4,9 +4,18 @@ import { DialogSurfaceElement } from '../DialogSurface';
 import type { Context } from '@fluentui/react-context-selector';
 import type { DialogModalType, DialogOpenChangeData } from '../Dialog';
 import { useModalAttributes } from '@fluentui/react-tabster';
+import { MotionState } from '@fluentui/react-motion-preview';
```

Comment:
Is this supposed to be just a draft or do you actually plan to merge it? We should never add a dependency on `-preview` package.

Reply:
The idea is to merge it eventually! I guess we could block the merging until the release of `@fluentui/react-motion-preview`

Reply:
is it about not having a `-preview` dependency or not having it leaking into our API surface? if the problem is leaking it to the API we can manage to obscure it...

Reply:
> Is this supposed to be just a draft or do you actually plan to merge it? We should never add a dependency on -preview package.

Is this the case? I don't see this in our docs Is this supposed to be just a draft or do you actually plan to merge it? We should never add a dependency on -preview package.

@Hotell  can you give us some clarity here?

Reply:
I see two problems:
1. The type from `-preview` package leaks into the API of the stable package. What are you going to do if that type changes or is removed completely.
2. We are introducing a caret dependency on a preview package into a stable package. Once the -preview package is bumped (with breaking changes in v0) we will end up dealing with the same dependency hell we faced with `react-components/unstable`.

TBH I feel this breaks the whole intent of the `-preview` package. Would love to hear @Hotell's opinion.

Reply:
Ok, going per parts:

1. problem 1 is quite easy to solve, we'd just need to obscure our API for now, using something like `unknown` or an equivalent type, to ensure we'll not leak a preview package into the API.
2. maybe we could introduce the dependency without a caret? WDYT @Hotell  @ling1726  @miroslavstastny